synapse/events/__init__.py

```diff
@@ -72,6 +72,10 @@ def __init__(self, event_dict, signatures={}, unsigned={},
 
         self._event_dict = event_dict
 
+        # some malformed events have no hashes. Avoid throwing exceptions
+        # later.
+        event_dict.setdefault("hashes", {})
```

Comment:
We really don't want to be editing the dict here, as it will invalidate any signatures